packages/sls-env/src/dependencies.ts

```diff
@@ -70,22 +95,39 @@ const resolveEventDependencies = <D, P, E, C>(
   context: C
 ): AppDependencyConverter<D> => {
   const updatedDependencies = Object.entries(dependencies).reduce((acc, [dependencyName, dependency]) => {
-    if (dependency?.type === 'EventDependency') {
-      const key = dependency.getKey(payload, event, context)
-      const eventDependency = dependency.dependencies[key]
-      if (!eventDependency) {
-        logger.warn(`could not extract event based dependency for ${dependencyName}`)
-        throw new EventDependencyError(`No event based dependency found for '${dependencyName}'`)
-      } else {
-        logger.debug(`replacing '${dependencyName}' with event based dependency`)
-        return { ...acc, [dependencyName]: eventDependency }
-      }
+    switch (dependency.type) {
+      case DependencyType.EventDependency:
+        const eventDependency = dependency as EventDependency<D, P, E, C>
+        const key = eventDependency.getKey(payload, event, context)
+        const resolvedEventDependency = eventDependency.dependencies[key]
+        if (!resolvedEventDependency) {
+          logger.warn(`could not extract event based dependency for ${dependencyName}`)
+          throw new EventDependencyError(`No event based dependency found for '${dependencyName}'`)
+        } else {
+          logger.debug(`replacing '${dependencyName}' with event based dependency`)
+          return { ...acc, [dependencyName]: resolvedEventDependency }
+        }
+      case DependencyType.DependencyFactory:
+        const dependencyFactory = dependency as DependencyFactory<D, P, E, C>
+        logger.debug(`replacing '${dependencyName}' with factory based dependency`)
+        try {
+          return { ...acc, [dependencyName]: dependencyFactory.resolveDependency({ payload, event, context }) }
+        } catch (error: unknown) {
+          throw new EventDependencyError((error as Error).message)
+        }
```

Comment:
`undefined` or `null` should return error but all other conditions should throw as normal

Reply:
could you please rename this to `DependencyNotFoundError`?